packages/x-scheduler/src/joy/agenda-view/AgendaView.tsx

```diff
@@ -0,0 +1,89 @@
+'use client';
+import * as React from 'react';
+import clsx from 'clsx';
+import { getAdapter } from '../../primitives/utils/adapter/getAdapter';
+import { useForkRef, useModernLayoutEffect } from '@base-ui-components/react/utils';
+import { AgendaViewProps } from './AgendaView.types';
+import { CalendarEvent } from '../models/events';
+import { useDayList } from '../../primitives/use-day-list/useDayList';
+import { AgendaEvent } from '../internals/components/event/agenda-event/AgendaEvent';
+import { useEventCalendarStore } from '../internals/hooks/useEventCalendarStore';
+import { useSelector } from '../../base-ui-copy/utils/store';
+import { selectors } from '../event-calendar/store';
+import { EventPopoverProvider } from '../internals/utils/EventPopoverProvider';
+import './AgendaView.css';
+
+const adapter = getAdapter();
+
+export const AgendaView = React.memo(
+  React.forwardRef(function AgendaView(
+    props: AgendaViewProps,
+    forwardedRef: React.ForwardedRef<HTMLDivElement>,
+  ) {
+    const containerRef = React.useRef<HTMLElement | null>(null);
+    const handleRef = useForkRef(forwardedRef, containerRef);
+
+    const { onEventsChange, className, ...other } = props;
+    const store = useEventCalendarStore();
+
+    const today = adapter.date();
+
+    const visibleDate = useSelector(store, selectors.visibleDate);
+
+    const getDayList = useDayList();
+
+    const days = React.useMemo(
+      () => getDayList({ date: visibleDate.startOf('week'), amount: 12 }),
+      [getDayList, visibleDate],
+    );
+
+    const getEventsStartingInDay = useSelector(store, selectors.getEventsStartingInDay);
+    const resourcesByIdMap = useSelector(store, selectors.resourcesByIdMap);
+
+    return (
+      <div ref={handleRef} className={clsx('AgendaViewContainer', 'joy', className)} {...other}>
+        <EventPopoverProvider containerRef={containerRef} onEventsChange={onEventsChange}>
+          {({ onEventClick }) => (
```

Comment:
@rita-codes not needed for now, but we could even have `EventPopoverProvider` expose `onEventClick` through a context that our various event component would consume (and probably rename is `openEditingPopover` so that the event component itself decides which interactions should start the edition (click, double click etc...).

That way we remove the render prop and everything is nicely encapsulated.